CONTRIBUTING.md

```diff
@@ -1,42 +1,56 @@
 We welcome your contributions! There are multiple ways to contribute.
 
-## Issues
-For bugs or enhancement requests, please file a GitHub issue unless it's security related. When filing a bug remember that the better written the bug is, the more likely it is to be fixed. If you think you've found a security vulnerability, do not raise a GitHub issue and follow the instructions on our [Security Policy](./SECURITY.md). 
+## Opening issues
 
-## Contributing code
+For bugs or enhancement requests, please file a GitHub issue unless it's
+security related. When filing a bug remember that the better written the bug is,
+the more likely it is to be fixed. If you think you've found a security
+vulnerability, do not raise a GitHub issue and follow the instructions in our
+[security policy](./SECURITY.md).
 
-We welcome your code contributions. To get started, you will need to sign the [Oracle Contributor Agreement](https://oca.opensource.oracle.com/) (OCA).
+## Contributing code
 
-For pull requests to be accepted, the bottom of your commit message must have
-the following line using the name and e-mail address you used for the OCA.
+We welcome your code contributions. Before submitting code via a pull request,
+you will need to have signed the [Oracle Contributor Agreement][OCA] (OCA) and
+your commits need to include the following line using the name and e-mail
+address you used to sign the OCA:
 
-```
+```text
 Signed-off-by: Your Name <you@example.org>
 ```
 
-This can be automatically added to pull requests by committing with:
+This can be automatically added to pull requests by committing with `--sign-off`
+or `-s`, e.g.
 
-```
+```text
 git commit --signoff
 ```
 
-Only pull requests from committers that can be verified as having
-signed the OCA can be accepted.
+Only pull requests from committers that can be verified as having signed the OCA
+can be accepted.
 
 ## Pull request process
 
-1. Fork this repository
+1. Ensure there is an issue created to track and discuss the fix or enhancement
+   you intend to submit.
+1. Fork this repository.
 1. Create a branch in your fork to implement the changes. We recommend using
-the issue number as part of your branch name, e.g., `1234-fixes`
-1. Ensure that any Javadoc and/or documentation is updated with the changes.
-1. Add a test for the new behaviour (or that exercises the bug if a bug fix).
-1. Submit the pull request. *Do not leave the pull request text blank*. Explain exactly
-what your changes are meant to do and provide simple steps on how to validate
-your changes. Ensure that you reference the issue you created as well. The PR 
-name will be the name of the squashed commit to main.
-1. We will assign the pull request to be reviewed before it is merged.
+   the issue number as part of your branch name, e.g. `1234-fixes`.
+1. Ensure that any documentation is updated with the changes that are required
+   by your change.
+1. Ensure that any samples are updated if the base image has been changed.
```

Comment:
Is this line necessary? We don't have a "base image", so maybe it should be "Ensure the tutorials are updated with any API changes"?

Reply:
updated